odoo/models.py

```diff
@@ -4269,6 +4255,78 @@ def _check_company(self, fnames=None):
                 })
             raise UserError("\n".join(lines))
 
+    def check_access(self, operation: str) -> None:
+        """ Verify that the current user is allowed to perform ``operation`` on
+        all the records in ``self``. The method raises an :class:`AccessError`
+        if the operation is forbidden on the model in general, or on any record
+        in ``self``.
+
+        In particular, when ``self`` is empty, the method checks whether the
+        current user has some permission to perform ``operation`` on the model
+        in general::
+
+            # check that user has some minimal permission on the model
+            records.browse().check_access(operation)
+
+        """
+        if not self.env.su:
+            self._check_access(operation, raise_exception=True)
+
+    def has_access(self, operation: str) -> bool:
+        """ Return whether the current user is allowed to perform ``operation``
+        on all the records in ``self``. The method is fully consistent with
+        method :meth:`check_access` but returns a boolean instead.
+        """
+        if not self.env.su and (forbidden := self._check_access(operation)):
+            forbidden.invalidate_recordset()
+            return False
+        return True
+
+    def _filter_access(self, operation: str):
+        """ Return the subset of ``self`` for which the current user is allowed
+        to perform ``operation``. The method is fully equivalent to::
+
+            self.filtered(lambda record: record.has_access(operation))
+
+        """
+        if self and not self.env.su and (forbidden := self._check_access(operation)):
+            forbidden.invalidate_recordset()
+            return self - forbidden
+        return self
+
+    def _check_access(self, operation: str, raise_exception: bool = False) -> Self:
+        """ Return the records in ``self`` on which the current user is not
+        allowed to perform ``operation``. In case ``self`` is empty and the
+        current user has no permission to perform ``operation`` on the model in
+        general, the method returns some arbitrary record instead. In other
+        words, the current user has the expected permissions if and only if the
+        result is empty.
+
+        If ``raise_exception`` is ``True``, the method raises an exception
+        instead of returning forbidden records.
+
+        This method provides the base implementation of
+        methods :meth:`check_access`, :meth:`has_access`
+        and :meth:`_filter_access`. The method may be overridden in order to
+        restrict the access to ``self``.
+        """
+        Access = self.env['ir.model.access']
+        if not Access.check(self._name, operation, raise_exception):
+            return self or self.browse(1)
```

Comment:
Or maybe anything else? Even if I didn't see an obvious bug, it seems like it's asking for it :smile: 
```suggestion
            return self or self.browse([False])
```

Reply:
Other functions expect a truthy value if you do not have access. It is falsy if you make this change.